purple_air.js

```diff
@@ -433,20 +444,39 @@ var purple_air = {
         let fetch_purple_air = await (await fetch(BASE_PURPLE_AIR_URL)).json()
 
         let data = fetch_purple_air.data
-        for (let d of data) {
```

Comment:
This new if else is adding redundant code as the same code is copied in both places. Consider below alternative solution below.  We can keep track of the number of sensors data reading and break the loop if max cap is reached.  Below solution doesn't duplicate the code. 

```
let totalReqSensors = purple_air.state.sensorNum || 0
let recordsCount = 0;

for (let d of data) {
    let newRow = { ...purple_air_fields }
    newRow.Location = purple_air.state.city
    newRow.sensor_index = d[0]
    newRow.name = d[1]
    // newRow.primary_id_a =   d[2]
    // newRow.primary_key_a =  d[3]
    newRow.latitude = d[2]
    newRow.longitude = d[3]
    // latLngList.push( `${d[2]},${d[3]}` )
    sensorValues.push(newRow)

    // Keep count of the sensor records read and terminate if max cap is reached
    recordsCount = recordsCount + 1; 
    if(totalReqSensors > 0 && recordsCount >= totalReqSensors){
    	break;
    }
}

```

Reply:
Thank you!